### PR TITLE
added a helper method in block entity init to help with block entity registration

### PR DESCRIPTION
**Pull request by xana43**
not sure if this should be here or in Turtylib

### CODE REVIEW

File: src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java [hunk 1]
```diff
@@ -12,6 +14,12 @@ public final class BlockEntityInit extends AbstractInit {
...
-    public static final RegistryObject<BlockEntityType<ClayAlloyFurnaceBlockEntity>> CLAY_ALLOY_FURNACE = BLOCK_ENTITIES.register("clay_alloy_furnace",
-            () -> BlockEntityType.Builder.of(ClayAlloyFurnaceBlockEntity::new, BlockInit.CLAY_ALLOY_FURNACE.get()).build(null));
+    public static final RegistryObject<BlockEntityType<ClayAlloyFurnaceBlockEntity>> CLAY_ALLOY_FURNACE = registerNewBlockEntity("clay_allow_furnace",ClayAlloyFurnaceBlockEntity::new,BlockInit.CLAY_ALLOY_FURNACE.get());
+
+
+
+    private static <T extends BlockEntity>RegistryObject<BlockEntityType<T>> registerNewBlockEntity(final String name, final BlockEntityType.BlockEntitySupplier<T> blockEntitySupplier, final Block block)
+    {
+        return BLOCK_ENTITIES.register(name,() -> BlockEntityType.Builder.of(blockEntitySupplier,block).build(null));
+    }
```
Comment (DaRealTurtyWurty) [inline src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java]:
```suggestion
    private static <T extends BlockEntity> RegistryObject<BlockEntityType<T>> registerNewBlockEntity(final String name, final BlockEntityType.BlockEntitySupplier<T> blockEntitySupplier, final Block... blocks) {
        return BLOCK_ENTITIES.register(name, () -> BlockEntityType.Builder.of(blockEntitySupplier, blocks).build(null));
    }
```

**Review by DaRealTurtyWurty — CHANGES_REQUESTED** [review]

File: src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java (hunk 1 above)
Comment (xana43) [reply src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java]: why a variable array of blocks?

Comment (DaRealTurtyWurty) [reply src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java]: Well because that's what its meant to be.

Comment (xana43) [reply src/main/java/io/github/darealturtywurty/turtychemistry/core/init/BlockEntityInit.java]: ooh ok

**Review by DaRealTurtyWurty — APPROVED** [review]

**Pull request merged**